src/ucp/api/ucp.h

```diff
@@ -131,15 +131,19 @@ enum ucp_params_field {
  * during @ref ucp_init "UCP initialization" process.
  */
 enum ucp_feature {
-    UCP_FEATURE_TAG    = UCS_BIT(0),  /**< Request tag matching support */
-    UCP_FEATURE_RMA    = UCS_BIT(1),  /**< Request remote memory
-                                           access support */
-    UCP_FEATURE_AMO32  = UCS_BIT(2),  /**< Request 32-bit atomic
-                                           operations support */
-    UCP_FEATURE_AMO64  = UCS_BIT(3),  /**< Request 64-bit atomic
-                                           operations support */
-    UCP_FEATURE_WAKEUP = UCS_BIT(4)   /**< Request interrupt notification
-                                           support */
+    UCP_FEATURE_TAG             = UCS_BIT(0), /**< Request tag matching support */
+    UCP_FEATURE_RMA             = UCS_BIT(1), /**< Request remote memory
+                                                   access support */
+    UCP_FEATURE_AMO32           = UCS_BIT(2), /**< Request 32-bit atomic
+                                                   operations support */
+    UCP_FEATURE_AMO64           = UCS_BIT(3), /**< Request 64-bit atomic
+                                                   operations support */
+    UCP_FEATURE_WAKEUP          = UCS_BIT(4), /**< Request interrupt notification
+                                                   support */
+    UCP_FEATURE_FAULT_TOLERANCE = UCS_BIT(5)  /**< Request fault tolerance support,
+                                                   may disable some API returning
```

Comment:
Regarding to disabling of some APIs. Really, I see only one - ucp_tag_send_sync_nb.
The reason is no way to detect remote failure between send and acknowledge in current implementation.
This limitation is temporary because we plan to implement "keep alive" protocol in future which should resolve this limitation but we have to have it documented.